source/gl/Texture.cpp

```diff
@@ -34,22 +34,22 @@ Texture::~Texture() noexcept {
 }
 
 void Texture::loadFromFile(const std::string &filename) {
-	SDL_Surface *surface = IMG_Load(filename.c_str());
-	if(!surface) {
+	sf::Image image;
+	if(!image.loadFromFile(filename)) {
 		throw EXCEPTION("Failed to load texture:", filename);
 	}
 
 	m_filename = filename;
 
-	m_width = surface->w;
-	m_height = surface->h;
+    m_width = image.getSize().x;
+    m_height = image.getSize().y;
 
 	glGenTextures(1, &m_texture);
 
 	bind(this);
 
-	GLenum format = (surface->format->BytesPerPixel == 4) ? GL_RGBA : GL_RGB;
-	glTexImage2D(GL_TEXTURE_2D, 0, format, m_width, m_height, 0, format, GL_UNSIGNED_BYTE, surface->pixels);
+	GLenum format = GL_RGBA;  //sf::Image is always stored as RGBA. Support for GL_RGB needed?
+	glTexImage2D(GL_TEXTURE_2D, 0, format, m_width, m_height, 0, format, GL_UNSIGNED_BYTE, image.getPixelsPtr());
```

Comment:
I think I'll use `sf::Texture` directly instead of using my own class, but if your changes here works properly then a change isn't really needed

Reply:
Yeah it works. I removed the comment and let you decide later what to do :)